File: P1/mario.c
#include <stdio.h>
#include <cs50.h>

//Mario pyramid generator program
int main(void)
{
    //Gets height of pyramid from user, prompts user till input is between 1-8 inclusive
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);

    //Prints n number of rows
    for (int i = 0; i < n; i++)
    {
        //Prints n-i-1 number of spaces at the beginning of each row
        for (int j = 0; j < n - i - 1; j++)
        {
            printf(" ");
        }

        //Prints i+1 number of #s for left side of pyramid
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }

        //Prints 2 spaces (constant)
        printf("  ");

        //Prints i+1 number of #s for right side of pyramid
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: P6/mario.py
from cs50 import get_int

n = 0
while n < 1 or n > 8:
    n = get_int("Height: ")

i = 0
while i < n:

    j = 0
    while j < (n - i - 1):
        print(" ", end="")
        j += 1

    k = 0
    while k < (i + 1):
        print("#", end="")
        k += 1

    print("  ", end="")

    l = 0
    while l < (i + 1):
        print("#", end="")
        l += 1

    print("")

    i += 1